### ISSUE TITLE
BatchLexer performce issue with multiple variables after if

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#963]
(Original issue 1256 created by roskakori on 2016-07-13T23:48:28.191896+00:00)

Here's is another performance issue in the ``shell.BatchLexer`` which has not been addressed by the fixes for #1216 and #1240. It can be reproduced with the following example code:

```python
def test_can_process_variable_after_if():
    lexer = lexers.get_lexer_by_name('bat')
    # text = 'if %suffix% equ .jar set LIBPATH=%LIBPATH%;"%REPO%\\%filename%"'
    text = 'if %xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx% %a%'
    for token in lexer.get_tokens(text):
        print(token)

if __name__ == '__main__':
    test_can_process_variable_after_if()
```
The more ``x`` there are in

```
if %xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx% %a%
```
the slower the lexer gets though eventually it will finish. While this is not valid batch code, the example includes an alternative real life code snipplet taken from the Neo4j launcher script:
```
if %suffix% equ .jar set LIBPATH=%LIBPATH%;"%REPO%\\%filename%"
```

**Comment by Anteru** [comment]
(Original issue 1256 last updated on 2017-01-22T17:45:25.599030+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by dscorbett on 2016-07-14T01:28:06.851126+00:00)

Fixed in pull request #620.

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2017-01-22T17:45:25.614643+00:00)

Merged!
